vital/algo/video_input.cxx

```diff
@@ -75,6 +75,15 @@ ::frame_rate()
 }
 
 
+// ------------------------------------------------------------------
+kwiver::vital::path_t
+video_input
+::filename() const
+{
+  return "";
```

Comment:
Please return a default-constructed string instead; it is likely more efficient... `return {}` is fine.